### PR TITLE
Tools: added Erle-Brain 2 build to CI 

### PR DESCRIPTION
**Pull request by LanderU**

### CODE REVIEW

**Comment by lucasdemarchi** [comment]
@LanderU adding all boards will probably make CI to stop compiling due to the long time. We should try to improve coverage by building almost everything, but building everything will probably not happen.

@OXINARF thoughts?

**Comment by lucasdemarchi** [comment]
same comment for #4058

**Comment by OXINARF** [comment]
@lucasdemarchi I asked for your thoughts in Gitter :smile:

Right now I don't think we have a problem with builds stopping due to time (I've seen builds longer than one hour), but this is undocumented behavior.
We can add a new job to add some more boards but Travis only provides 5 concurrent jobs (and we already have 6 jobs per test). I'm not sure if there's any interest in talking to Travis to try to get more concurrent jobs or not, may be this is a discussion for the dev call?

I believe it's important to test every board we can because we have different code for them but we have limited resources.

File: .travis.yml [hunk 1]
```diff
@@ -66,12 +66,12 @@ matrix:
     - compiler: "gcc PX4"
       env: CI_BUILD_TARGET="px4-v2 px4-v4"
     - compiler: "gcc Native"
-      env: CI_BUILD_TARGET="sitl linux minlure"
+      env: CI_BUILD_TARGET="erlebrain2 sitl linux minlure"
     - compiler: "gcc Linux"
-      env: CI_BUILD_TARGET="navio2 raspilot bebop"
+      env: CI_BUILD_TARGET="erlebrain2 navio2 raspilot bebop"
```
Comment (OXINARF) [inline .travis.yml]: It should either be in _gcc Native_ or _gcc Linux_, not both

**Comment by lucasdemarchi** [comment]
Ok, assigning to you. Thanks.

**Comment by kwikius** [comment]
How about creating a separate branch per board solely for the purpose of testing and send each to Travis as a separate job?

**Comment by OXINARF** [comment]
There's no point in that. The 5 concurrent jobs limit is per account, not per branch nor even per repository.

**Comment by kwikius** [comment]
Send them as a sequence ?
Send Part N dependent on part N-1 succeeding...

**Comment by kwikius** [comment]
Actually it is quite easy. Dont do the clang builds unless the gcc builds pass. The front end of both is very similar. That frees up 2 slots. Why do the Linux builds take so long anyway. I would have thought you could use prebuilt libraries? Just build what has changed ?

**Comment by kwikius** [comment]
Linux heavy though aint it LOL :0

**Comment by OXINARF** [comment]
I'm not going to explain how Travis works (they have good documentation if you want to read it), but what you are saying doesn't work, we don't send nothing, Travis automatically builds branches and PRs. Also your proposal would make it even slower, waiting for previous jobs to complete before starting new ones.

**Comment by kwikius** [comment]
Travis builds what you tell it to in the build script  ;) 
No Brainer  Tests must be be built for all boards. All boards must be built. All examples must be built. else remove them from the codebase. The issue isnt about speed , its about coverage of all boards as well as tests and examples for all boards.

**Comment by OXINARF** [comment]
> Travis builds what you tell it to in the build script ;) 

Exactly, and there's no way to tell it to wait to start one job only if another completed.

> No Brainer. Tests must be be built for all boards. All boards must be built. All examples must be built for all boards, else remove them from the codebase. The issue isnt about speed , it's about coverage of all boards as well as tests and examples for all boards.

Said who? Yes, we have an issue with speed, waiting hours to get a PR tested isn't acceptable.

**Comment by kwikius** [comment]
So you are happy to accept code that hasnt been tested ?  Is a fast build more important than correct code?

**Comment by OXINARF** [comment]
Code is reviewed before going in. If it is specific to a board (which would be the only issue we have) then the maintainer for that board (usually the creator of the board) is asked to review it. We also do extensive beta-testing before a release.
Yes, like I said before, I would love to test all boards, but, like I also said before, we don't have unlimited resources. This is really a matter for a dev call, which you are welcome to attend.

**Comment by vmayoral** [comment]
Thanks @lucasdemarchi and @OXINARF for looking into this. Having commits tested for our boards and binaries generated at firmware.ardupilot.org is something many would appreciate. I understand resources are limited but given the popularity of Erle-Brain 2 and PXFmini I believe it's a fair request. Is there a chance this PR and #4058 can get in?

If it helps, we can probably remove the PXF (discontinued part) from the setup to free some resources. Our team would be happy to continue supporting those that still work with PXF through [our forum](http://forum.erlerobotics.com/c/brains/erle-brain).

**Comment by OXINARF** [comment]
@vmayoral I'm not sure if I'll be able to attend the next dev call, but if I am I will discuss this and we'll take a decision there. Generating binaries is probably more acceptable because there's not usually an immediate need to get it, the issue is to wait too much to get PRs tested.

**Comment by lucasdemarchi** [comment]
As long as build doesn't start timing out in travis (and thus failing for other PRs). Once the comment from @OXINARF is fixed I think we can get these in.

**Comment by OXINARF** [comment]
@lucasdemarchi As it is I wouldn't want it in. I prefer we create a new job in Travis for these boards - as it is it would also impact our ccache.

**Comment by lucasdemarchi** [comment]
@OXINARF agreed.

**Comment by OXINARF** [comment]
@LanderU We talked about this in the dev call and the decision is to start using Semaphore to test different boards instead of the duplication we have now.
If you can just leave the build_binaries.sh commit it would be great or **whoever gets this in should just merge the build_binaries commit**
I'll bring a PR, hopefully tomorrow, to make the changes necessary to test this board.

**Comment by lucasdemarchi** [comment]
Applied the change to the build_binaries.sh script. @OXINARF will take a look in the rest.

**Comment by OXINARF** [comment]
@LanderU I've opened the PR #4097 to simplify the CI build script and, after that goes in, I will change the Semaphore targets to include erlebrain2.

**Pull request closed without merging**